src/metabase/search/api.clj

```diff
@@ -63,7 +63,7 @@
   (api/check-superuser)
   (if  (search/supports-index?)
     ;; The job appears to wait on the main thread when run from tests, so, unfortunately, testing this branch is hard.
-    (if (and (task/job-exists? task.search-index/reindex-job-key) (not config/is-test?))
+    (if (and (task/job-exists? task.search-index/reindex-job-key) (not ingestion/*force-sync*))
```

Comment:
We need this condition to hold whenever we're using the test running, even if that particular test has not forced sync for *all* mutations.

Perhaps this works in practice for now, but it's pretty brittle. I'm not sure of a general way to test whether you're inside a test besides checking the current thread name, but adding that with an `or` is better than nothing I guess.